bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -5938,6 +6041,189 @@ private static void closeStatement(Statement statement) {
+			throw new TskCoreException("Error updating the end date.", ex);
+		} finally {
+			connection.close();
+			releaseSharedLock();
+		}
+	}
+
+	/**
+	 * Adds the given ingest job to the database
+	 *
+	 * @param dataSource    The datasource the ingest job is being run on
+	 * @param hostName      The name of the host
+	 * @param ingestModules The ingest modules being run during the ingest job
+	 * @param jobStart      The time the job started
+	 *
+	 * @return An information object representing the ingest job added to the
+	 *         database.
+	 *
+	 * @throws TskCoreException If adding the job to the database fails.
+	 */
+	public final IngestJobInfo addIngestJob(Content dataSource, String hostName, List<IngestModuleInfo> ingestModules, Date jobStart) throws TskCoreException {
+		CaseDbConnection connection = connections.getConnection();
+		acquireSharedLock();
+		ResultSet resultSet = null;
+		Statement statement = null;
+		try {
+			statement = connection.createStatement();
+			String update = "INSERT INTO ingest_jobs (data_src_id, host_name, start_date_time, end_date_time, status_id, settings_dir) "
+					+ "VALUES (" + dataSource.getId() + ", '" + hostName + "', " + jobStart.getTime() + ", 0, 0, '');";
+			statement.executeUpdate(update);
+			resultSet = statement.getGeneratedKeys();
+			resultSet.next();
+			long id = resultSet.getLong(1);
+			for (int i = 0; i < ingestModules.size(); i++) {
+				IngestModuleInfo ingestModule = ingestModules.get(i);
+				statement.executeUpdate("INSERT INTO ingest_job_modules (ingest_job_id, ingest_module_id, pipeline_position) "
+						+ "VALUES (" + id + ", " + ingestModule.getIngestModuleId() + ", " + i + ");");
+			}
+			resultSet.close();
+			resultSet = null;
+			return new IngestJobInfo(id, dataSource.getId(), hostName, jobStart, "", ingestModules, this);
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error adding the ingest job.", ex);
+		} finally {
+			closeResultSet(resultSet);
+			connection.close();
+			releaseSharedLock();
+		}
+	}
+
+	/**
+	 * Adds the given ingest module to the database.
+	 *
+	 * @param displayName The display name of the module
+	 * @param uniqueName  The unique name of the module. Must be unique to this
+	 *                    module.
+	 * @param type        The type of the module.
+	 * @param version     The version of the module.
+	 *
+	 * @return An ingest module info object representing the module added to the
+	 *         db.
+	 *
+	 * @throws TskCoreException When the ingest module cannot be added.
+	 */
+	public final IngestModuleInfo addIngestModule(String displayName, String uniqueName, IngestModuleType type, String version) throws TskCoreException {
+		CaseDbConnection connection = connections.getConnection();
+		ResultSet resultSet = null;
+		Statement statement = null;
+		try {
+			statement = connection.createStatement();
+			resultSet = statement.executeQuery("SELECT * FROM ingest_modules WHERE unique_name = '" + uniqueName + "'");
```

Comment:
The SELECT here is vulnerable to race conditions for a multi-user case - a node might add the ingest module between the SELECT and the INSERT. We will need to try another SELECT if the INSERT throws due to violation of the unique constraint on the unique_name column.